util/weeklog.js

```diff
@@ -26,7 +27,9 @@ function parseLog (buffer, categoryList) {
     for (let j = 0; j < 8; j ++) {
       steamid += BigInt(buffer[curr + j]) * BigInt(Math.pow(256, 7 - j));
     }
+    // convert the acquired number to a valid steamid string
     entry.steamid = steamid.toString();
+    while (entry.steamid.length < 17) entry.steamid = "0" + entry.steamid;
```

Comment:
This seems incredibly scuffed.. what if the length is not one, but two characters short? Also, how did this piece of code work before this line as put in place?

Reply:
It's a while loop, so it'll leftpad until the length is 17.

It worked before because we've never attempted to use week logs with forged SteamIDs. The currently deployed system has some placeholder values for runners that came and went before we had Steam auth, and those IDs are basically a bunch of zeroes with an index at the end. The log used to parse this as a number, we need a string from it. I only now noticed that this is a problem, because there weren't many cases where you'd pass the SteamID from an ancient log into a util, until now.

Reply:
> It's a while loop

In my defense, I am eepy